lib/board.rb

```diff
@@ -0,0 +1,66 @@
+class Board
+  attr_accessor :cells
+
+    def initialize
+      @cells = Array.new(9, " ")
+    end
+
+    def reset!
+      self.cells.clear
+        self.cells = Array.new(9, " ")
+    end
+
+    def display
+      puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+      puts "-----------"
+      puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+      puts "-----------"
+      puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+    end
+
+    def position(location)
+      new_location = location.to_i - 1
+        self.cells[new_location]
+    end
+
+    def full?
+      self.cells.all? { |token| token == "X" || token == "O" }
+    end
+
+    def turn_count
+      counter = 0
+      self.cells.each do |current_move|
+      if current_move == 'X' || current_move == 'O'
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```